buildstockbatch/aws/aws.py

```diff
@@ -22,22 +25,54 @@
 import pathlib
 import random
 from s3fs import S3FileSystem
+import shutil
 import tarfile
 import re
+import tempfile
 import time
+import tqdm
 import io
-import zipfile
+import yaml
 
-from buildstockbatch.aws.awsbase import AwsJobBase
 from buildstockbatch.base import ValidationError
+from buildstockbatch.aws.awsbase import AwsJobBase, boto_client_config
 from buildstockbatch.cloud.docker_base import DockerBatchBase
-from buildstockbatch.utils import log_error_details, get_project_configuration
+from buildstockbatch.utils import (
+    log_error_details,
+    get_project_configuration,
+    get_bool_env_var,
+)
 
 logger = logging.getLogger(__name__)
 
 
+def backoff(thefunc, *args, **kwargs):
```

Comment:
This backoff handles other retries, like if it throws an error because you're trying to modify something you just created like add a network interface to a VPC or something.